### PR TITLE
Store descriptions in the scripts property

### LINKED ISSUE #6: Adding scripts descriptions to the scripts property
IMHO, the descriptions in a different section are a little bit hard to read/edit.

I think it would be nice to add the possibility to write the descriptions in the `scripts` property by prefixing the script name with a `?`

For example, instead of this

``` json
{
  "name": "my-project",
  "scripts": {
    "info": "npm-scripts-info"
  },
  "scripts-info": {
    "info": "Displays information about the scripts.",
    "watch:build": "Compiles the scripts and watches for changes.",
    "start": "Kickstarts the application."
  }
}
```

This can be written

``` json
{
  "name": "my-project",
  "scripts": {
    "info": "npm-scripts-info",
    "?info": "Displays information about the scripts.",
    "?watch:build": "Compiles the scripts and watches for changes.",
    "?start": "Kickstarts the application."
  }
}
```

Furthermore, if the description will start with `echo`, it'll be possible to view it through npm run. Thus having this package.json

``` json
{
  "name": "my-project",
  "scripts": {
    "info": "npm-scripts-info",
    "?info": "echo Displays information about the scripts.",
    "?watch:build": "echo Compiles the scripts and watches for changes.",
    "?start": "echo Kickstarts the application."
  }
}
```

`npm run ?start` will output the description of the `start` script to the console. `npm-scripts-info` can use the same description and just remove the `echo` part.

I can implement this new feature if you like it.

### PR DESCRIPTION
**Pull request by zkochan**
Add the possibility to store the descriptions not only in the `scripts-info` property but in the `scripts` property as well. Properties that are prefixed with ? in the `scripts` are descriptions for the corresponding scripts.

closes #6

### CODE REVIEW

**Comment by srph** [comment]
Thanks. Will get back to this PR after I take a nap

**Comment by zkochan** [comment]
OK, cool

File: lib/index.js [hunk 1]
```diff
@@ -27,4 +27,40 @@ function info() {
...
+}
+
+function getScriptDecription(description) {
+  if (description.indexOf('echo ') !== 0) {
+    return description;
+  }
+
+  var woEchoDescription = description.replace('echo ', '');
+
+  return removeQuotes('"', removeQuotes("'", woEchoDescription));
```
Comment (srph) [inline lib/index.js]: I think you should use a trim function instead.

File: lib/index.js [hunk 2]
```diff
@@ -27,4 +27,40 @@ function info() {
...
+    });
+
+  return scriptsInfo;
+}
+
+function scriptIsDescription(scriptName) {
+  return scriptName[0] === '?';
+}
+
+function getScriptDecription(description) {
```
Comment (srph) [inline lib/index.js]: Typo, should be "De**s**cription". IMO, this is better named as `getDescription`

File: lib/index.js [hunk 3]
```diff
@@ -27,4 +27,40 @@ function info() {
...
+  Object.keys(pkgJSON.scripts || {})
+    .filter(scriptIsDescription)
+    .forEach(function(scriptName) {
+      scriptsInfo[scriptName.substr(1)] = getScriptDecription(pkgJSON.scripts[scriptName]);
+    });
+
+  return scriptsInfo;
+}
+
+function scriptIsDescription(scriptName) {
```
Comment (srph) [inline lib/index.js]: imo, `isDescription` would be a better name.

**Comment by zkochan** [comment]
Thanks for checking the PR. I agree with all of your comments and I have updated the branch.

**Comment by srph** [comment]
LGTM. Thanks a bunch! :100:

**Pull request merged**